File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        low=0
        right=len(nums)-1
        mid=0
        while(mid<=right):
            if nums[mid]==0:
                nums[low],nums[mid]=nums[mid],nums[low]
                low+=1
                mid+=1
            
            elif nums[mid]==1:
                mid+=1
            
            else:
                nums[right],nums[mid]=nums[mid],nums[right]
                right-=1

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    private static void swap(int l,int r,int[] nums){
        int temp=nums[l];
        nums[l]=nums[r];
        nums[r]=temp;
    }
    public void sortColors(int[] nums) {
        int l=0;
        int m=0;
        int r=nums.length-1;
        while(m<=r){
            if(nums[m]==2){
                swap(m,r,nums);
                r--;
            }
            else if(nums[m]==0){
                swap(l,m,nums);
                l++;
                m++;

            }
            else{
                m++;
            }
        }
    }
}